lua/entities/hl_camera/init.lua

```diff
@@ -128,13 +129,66 @@ function ENT:SetPlayerKey(ply, key, toggle)
 	self:UpdatePlayer(ply)
 end
 
+function ENT:GhostCreate()
+	if IsValid(self.ACGhost) then
+		return
+	end
+	local ghost = ents.Create("hl_camera_ghost")
+
+	ghost:Spawn()
+
+	ghost:SetParent(self)
+	ghost:SetLocalPos(self:GetViewOffset())
+	ghost:SetLocalAngles(Angle(0, 0, self:GetRoll()))
+
+	return ghost
+end
```

Comment:
If `self:GetViewOffset()` is the zero vector, you should NOT create a ghost. In that case, just return `nil` before calling `ents.Create`. I think the rest of the code would handle that just fine.